opm/simulators/linalg/cuistl/detail/vector_operations.cu

```diff
@@ -45,6 +47,60 @@ namespace
         }
     }
 
+    template <class T>
+    __global__ void weightedDiagMVBlocksize1(
+        T* squareBlockVector, const size_t numberOfElements, const T w, const T* globalSrcVec, T* globalDstVec)
+    {
+        const auto globalIndex = blockDim.x * blockIdx.x + threadIdx.x;
+        const int blocksize = 1;
+
+        if (globalIndex < numberOfElements) {
+            T* pMat = (squareBlockVector + (blocksize * blocksize * globalIndex));
+            const T* localSrcVec = (globalSrcVec + (blocksize * globalIndex));
+            T* localDstVec = (globalDstVec + (blocksize * globalIndex));
+
+            T v0 = localSrcVec[0];
+            localDstVec[0] = (pMat[0] * v0) * w;
+        }
+    }
+
+    template <class T>
+    __global__ void weightedDiagMVBlocksize2(
+        T* squareBlockVector, const size_t numberOfElements, const T w, const T* globalSrcVec, T* globalDstVec)
+    {
+        const auto globalIndex = blockDim.x * blockIdx.x + threadIdx.x;
+        const int blocksize = 2;
+
+        if (globalIndex < numberOfElements) {
+            T* pMat = (squareBlockVector + (blocksize * blocksize * globalIndex));
+            const T* localSrcVec = (globalSrcVec + (blocksize * globalIndex));
+            T* localDstVec = (globalDstVec + (blocksize * globalIndex));
+
+            T v0 = localSrcVec[0], v1 = localSrcVec[1];
```

Comment:
I'd break this out into two lines, see https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-stmt and https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl21-declare-one-name-only-per-declaration 

The same applies for line 97 n the blocksize=3-version.